packages/salesforcedx-apex-debugger/test/integration/adapter.test.ts

```diff
@@ -5,32 +5,229 @@
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
+import {
+  CliCommandExecutor,
+  CommandExecution,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import * as util from '@salesforce/salesforcedx-utils-vscode/out/src/test/orgUtils';
 import { expect } from 'chai';
+import * as path from 'path';
+import * as rimraf from 'rimraf';
+import * as Url from 'url';
 import { DebugClient } from 'vscode-debugadapter-testsupport';
+import { DebugProtocol } from 'vscode-debugprotocol';
+import { LaunchRequestArguments } from '../../src/adapter/apexDebug';
+import { LineBreakpointInfo } from '../../src/breakpoints/lineBreakpoint';
+import { LINE_BREAKPOINT_INFO_REQUEST } from '../../src/constants';
 
-describe('Debugger adapter - integration', () => {
-  describe('Session', () => {
-    let dc: DebugClient;
+const PROJECT_NAME = `project_${new Date().getTime()}`;
+const SIMPLE_VARIABLES_DIR = path.join(
+  __dirname,
+  '..',
+  '..',
+  '..',
+  'test',
+  'integration',
+  'config',
+  'variables'
+);
+const SOURCE_FOLDER = path.join(SIMPLE_VARIABLES_DIR, 'source');
+const APEX_EXEC_FILE = path.join(SIMPLE_VARIABLES_DIR, 'apexExec', 'test.apex');
+const LINE_BREAKPOINT_INFO: LineBreakpointInfo[] = [];
 
-    before(async () => {
-      dc = new DebugClient('node', './out/src/adapter/apexDebug.js', 'apex');
-      // Use dc.start(4711) to debug the adapter during
-      // tests (adapter needs to be launched in debug mode separately).
-      await dc.start();
+/**
+ * These integration tests assume the environment has authenticated to
+ * a Dev Hub and it is set as the default Dev Hub.
```

Comment:
If the tests require an authenticated DevHub it should be moved to system tests.

Reply:
On a side note, would it make sense to introduce a second kind of integration tests which deal with a "mocked" Apex server? But that might just be too much effort.

Reply:
@guw We've discussed that these integration tests don't fit the description of system tests. So, I've added a config in launch.json and script in package.json to run debugger unit and integration tests separately for our devs. Is there a specific flow that you're looking for in the autobuild?

Reply:
Thanks @DatGuyJonathan! Were you able to sort out the coverage issue? Re: flow I think we should discuss whether we want to run them by default with "npm run tests" or create a separate goal for "tests-with-devhub" in all packages. They should run after unit tests.

Reply:
@guw What do you have in mind from segmenting unit tests, integration tests with devhub, and UI tests in appveyor when currently we just run them all in appveyor.yml?

Reply:
I usually run the unit test locally before committing without a fully running server and authenticated DevHub available.

Reply:
Okay, I believe you should be able to do that with the changes in debugger's launch.json and package.json. We'll address the same concern with sobject tests in W-4402142.

Reply:
I experimented with making Appveyor run unit tests first, then integration tests, then system tests in branch jwidjaja/appveyor. It's not ready, but you check it out there. I'll merge this PR to develop for now.